### PR TITLE
Enable shortened methods by default

### PR DESCRIPTION
**Pull request by maxhaton**
The preview flag will still work but will print a deprecation message

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Review by Geod24 — CHANGES_REQUESTED** [review]
1) Don't deprecate the flag right away. If someone supports multiple versions, they'll have to check the version of DMD in their build script if they want a clean build. That is annoying and doesn't scale.
2) We need to be MUCH BETTER at explaining changes. And the one line, long title in the changelog is definitely not cutting it.

File: changelog/shortenedMethodsEnabled.dd [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Shortened methods are now enabled by default, that is to say no preview flag
```
Comment (Geod24) [inline changelog/shortenedMethodsEnabled.dd:1]: 1) Title should stay under 80 chars
2) Wording used (that is to say) is not professional
3) Add a description of the feature. When was it introduced ? Which DIP ? What does it do ?

Comment (maxhaton) [reply changelog/shortenedMethodsEnabled.dd:1]: Why is it not professional?

You may mean legalese but the wording is completely fine in any professional context.

**Comment by maxhaton** [comment]
made it undeprecated

how many releases do we keep them around for

**Review by ljmf00 — APPROVED** [review]

File: changelog/shortenedMethodsEnabled.dd (hunk 1 above)
Comment (maxhaton) [reply changelog/shortenedMethodsEnabled.dd:1]: will add something to changelog later, ran out of successful keypresses today.

Comment (dkorpel) [reply changelog/shortenedMethodsEnabled.dd:1]: Suggestion for a better changelog entry:
```
Shortened method syntax is now enabled by default

Following the acceptance of [DIP 1043---Shortened Method Syntax](https://forum.dlang.org/thread/ahswpyijfidyorixgaqm@forum.dlang.org), you no longer need to use the `-preview=shortenedMethods` flag to use shortened function bodies:

---
int add(int x, int y) pure => x + y;

// equivalent full function body:
int add(int x, int y) pure { return x + y; }
---

The now obsolete preview flag will be deprecated in a future release.
```

**Comment by dkorpel** [comment]
@maxhaton Can you update the changelog so we can get this in? See my suggestion (https://github.com/dlang/dmd/pull/14462#discussion_r977569439)

**Comment by maxhaton** [comment]
@Geod24 changelog 4 u

**Comment by Geod24** [comment]
```
ERROR: Newly generated header file (/media/disk1/braddr/sandbox/at-client/pull-5344065-Linux_32_64/dmd/generated/linux/release/64/frontend.h) doesn't match with the reference header file (/media/disk1/braddr/sandbox/at-client/pull-5344065-Linux_32_64/dmd/compiler/src/dmd/frontend.h)

DETAILS:

diff --git a/media/disk1/braddr/sandbox/at-client/pull-5344065-Linux_32_64/dmd/compiler/src/dmd/frontend.h b/media/disk1/braddr/sandbox/at-client/pull-5344065-Linux_32_64/dmd/generated/linux/release/64/frontend.h
index 1626123..32353a1 100644
--- a/media/disk1/braddr/sandbox/at-client/pull-5344065-Linux_32_64/dmd/compiler/src/dmd/frontend.h
+++ b/media/disk1/braddr/sandbox/at-client/pull-5344065-Linux_32_64/dmd/generated/linux/release/64/frontend.h
@@ -628,7 +628,7 @@ struct Param final
         noSharedAccess(),
         previewIn(),
         inclusiveInContracts(),
-        shortenedMethods(),
+        shortenedMethods(true),
         fixImmutableConv(),
         fix16997(true),
         useInvariants((CHECKENABLE)0u),

===============
The file `src/dmd/frontend.h` seems to be out of sync. This is likely because
changes were made which affect the C++ interface used by GDC and LDC.

```

**Review by Geod24 — APPROVED** [review]
LGTM once the CI agrees.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**